nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -79,6 +80,17 @@ in {
       default = "/var/lib/nextcloud";
       description = "Storage path of nextcloud.";
     };
+    datadir = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Data storage path of nextcloud. Will be services.nextcloud.home by default. This folder will be populated with a config and data folder which contain the state of the instance (excl the database).";
```

Comment:
This should really be a multi-line description IMHO.
Also I'd prefer to have options referenced like `<xref linkend="opt-services.nextcloud.home" />`